src/VCR.ts

```diff
@@ -12,8 +12,8 @@ interface VCROptions {
 export async function startVCR(options?: Partial<VCROptions>): Promise<nock.Scope[]> {
   const { processDefns, CI } = { processDefns: defns => defns, CI: process.env.CI, ...options } as VCROptions
   const defns = await currentCassettes()
-  if (!defns.length) {
-    if (CI) {
+  if (typeof defns === 'undefined' || !defns.length) {
+    if (CI && typeof defns === 'undefined') {
```

Comment:
Could we simplify this logic by making the CI check a guard like this?
```
if (CI && typeof defns === 'undefined') {
  throw new Error(`No cassettes found. They must be in place before running tests on CI ${cassettePath()}`)
}

if (defns?.length) {
  // Cassettes found - using saved responses
  // set up the mocks
  const scopes = nock.define(processDefns(defns.map(relaxDefinitionsForJsonRPC)))
  scopes.forEach(matchJsonRPCResponseToRequest)
  if (!nock.isActive()) nock.activate()
  return scopes
} else {
  // No cassettes found - recording responses
  nock.recorder.rec({ output_objects: true, dont_print: true })
}
```

Reply:
To do this I needed to rewrite the other ifs. I agree it makes sense. I don't like nested "ifs"
but it's done